core/src/subgraph/registrar.rs

```diff
@@ -668,10 +679,79 @@ fn create_subgraph_version(
         },
     });
 
+    let start_blocks: Option<Vec<EthereumBlockPointer>> = manifest
+        .data_sources
+        .clone()
+        .into_iter()
+        .map(|data_source| {
+            let source_address = data_source.source.address;
+            let block_number = match data_source
+                .source
+                .start_block
+                .and_then(|block_str| block_str.parse::<u64>().ok())
+            {
+                Some(number) => number,
+                None => {
+                    warn!(
+                        logger,
+                        "Start block invalid or not specified for the '{}' data source",
+                        &data_source.name
+                    );
+                    return None
+                },
+            };
+
+            // Validate that the start block provided for the Data Source is indeed the block that
+            // contains the contract creation transaction. Allow the block to be used as the
+            // subgraph level start block regardless, but emit a warning.
+            match ethereum_adapter
+                .validate_start_block(logger, block_number, source_address)
+                .wait()
+            {
+                Ok((block, validated)) => {
+                    if !validated {
+                        warn!(
+                            logger,
+                            "Start block for '{}' may lead to skipping on chain data, block number: {}, address: {}",
+                            &data_source.name,
+                            &block_number.to_string(),
+                            source_address
+                                .map(|a| a.to_string())
+                                .unwrap_or("unspecified".to_string())
+                        );
+                    }
+                    Some(block)
+                },
+                Err(e) => {
+                    warn!(
+                        logger,
+                        "Unable to validate start block for '{}', block_num: {}, error: {}",
+                        &block_number.to_string(),
+                        &data_source.name,
+                        e
+                    );
+                    None
+                }
+            }
+        })
+        .filter(|&n| n.is_some())
+        .collect();
+
     // Create deployment only if it does not exist already
     if !deployment_exists {
         let chain_head_block = chain_store.chain_head_ptr()?;
-        let genesis_block = chain_store.genesis_block_ptr()?;
+        let start_blocks = start_blocks.unwrap();
+        let genesis_block = if start_blocks.is_empty() {
```

Comment:
Since this variable might not be the genesis block, lets rename it to something else.